src/passes/bytesConverter.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  ArrayType,
+  ArrayTypeName,
+  Assignment,
+  BinaryOperation,
+  DataLocation,
+  ElementaryTypeName,
+  FixedBytesType,
+  FunctionCall,
+  FunctionDefinition,
+  getNodeType,
+  Identifier,
+  IndexAccess,
+  IntType,
+  Literal,
+  MemberAccess,
+  UnaryOperation,
+  typeNameToTypeNode,
+  VariableDeclaration,
+} from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+import { getFunctionTypeString } from '../utils/utils';
+
+/* Convert fixed-size byte arrays (e.g. bytes2, bytes8) to their equivalent unsigned integer.
+		This pass currently does not handle dynamically-sized bytes arrays (i.e. bytes).
+*/
+
+export class BytesConverter extends ASTMapper {
+  // Returns the integer value in string for a literal of integer constant type.
+  // Fixed-size bytes arrays declarations are literals of integer constant type, but with hex values.
+  replacementLiteralIntValue(node: Literal): string {
+    const literalInt = BigInt(node.value);
+    return literalInt.toString();
+  }
+
+  replacementIntTypeNode(typeNode: FixedBytesType): IntType {
+    return new IntType(typeNode.size * 8, false, undefined);
+  }
+
+  visitAssignment(node: Assignment, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
```

Comment:
This is a good approach for scalar fixed-bytes-types, but what about arrays of fixed bytes types?